graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrMutation.java

```diff
@@ -43,15 +43,39 @@
 @GraphQLDescription("JCR Mutations")
 public class GqlJcrMutation extends GqlJcrMutationSupport implements DXGraphQLFieldCompleter {
 
-    private String workspace;
+    private final NodeQueryExtensions.Workspace workspace;
     private boolean save = true;
 
     /**
      * Initializes an instance of this class with the specified JCR workspace name.
      *
      * @param workspace the name of the JCR workspace
+     * @deprecated use {@link #GqlJcrMutation(NodeQueryExtensions.Workspace, boolean)} instead
      */
     public GqlJcrMutation(String workspace, boolean save) {
+        this.workspace = getWorkspaceFromName(workspace);
+        this.save = save;
+    }
+
+    private NodeQueryExtensions.Workspace getWorkspaceFromName(String workspace) {
+        if (workspace == null) {
+            return null;
+        }
+        if (NodeQueryExtensions.Workspace.EDIT.toString().equalsIgnoreCase(workspace)) {
+            return NodeQueryExtensions.Workspace.EDIT;
+        }
+        if (NodeQueryExtensions.Workspace.LIVE.toString().equalsIgnoreCase(workspace)) {
+            return NodeQueryExtensions.Workspace.LIVE;
+        }
+        return null;
+    }
```

Comment:
(Even if used by the deprecated code only)t
his code logically deserve to be a static method inside the `NodeQueryExtensions.Workspace` class itself.
aka: `NodeQueryExtensions.Workspace.fromName(String)`

Reply:
I actually kept it here to ensure it would get removed when removing the deprecated constructor... 😅 
But it might actually be useful in the future so makes sense to be in `NodeQueryExtensions.Workspace`!